sycl/test/basic_tests/property_traits.cpp

```diff
@@ -0,0 +1,181 @@
+
+  // Context is_property_v
+  CHECK_IS_PROPERTY_V(property::context::cuda::use_primary_context);
+
+  // Image is_property_v
+  CHECK_IS_PROPERTY_V(property::image::use_host_ptr);
+  CHECK_IS_PROPERTY_V(property::image::use_mutex);
+  CHECK_IS_PROPERTY_V(property::image::context_bound);
+
+  // Queue is_property_v
+  CHECK_IS_PROPERTY_V(property::queue::in_order);
+  CHECK_IS_PROPERTY_V(property::queue::enable_profiling);
+  CHECK_IS_PROPERTY_V(property::queue::cuda::use_default_stream);
+
+  // Accessor is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::noinit,
+                         accessor<float, 1, access_mode::write, target::device,
+                                  access::placeholder::true_t>);
+  CHECK_IS_PROPERTY_OF_V(
+      property::no_init,
+      accessor<unsigned long, 2, access_mode::read, target::host_buffer,
+               access::placeholder::true_t>);
+  CHECK_IS_PROPERTY_OF_V(ext::oneapi::property::no_offset,
+                         accessor<long, 3, access_mode::read_write,
+                                  target::local, access::placeholder::false_t>);
+  CHECK_IS_PROPERTY_OF_V(ext::oneapi::property::no_alias,
+                         accessor<bool, 1, access_mode::read, target::device,
+                                  access::placeholder::true_t>);
+  CHECK_IS_PROPERTY_OF_V(
+      ext::intel::property::buffer_location,
+      accessor<sycl::half, 2, access_mode::write, target::host_buffer,
+               access::placeholder::false_t>);
+
+  // Host-ccessor is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::noinit,
+                         host_accessor<float, 1, access_mode::write>);
+  CHECK_IS_PROPERTY_OF_V(property::no_init,
+                         accessor<unsigned long, 2, access_mode::read>);
+
+  // Buffer is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::buffer::use_host_ptr, buffer<int, 2>);
+  CHECK_IS_PROPERTY_OF_V(property::buffer::use_mutex, buffer<char, 1>);
+  CHECK_IS_PROPERTY_OF_V(property::buffer::context_bound, buffer<float, 1>);
+  CHECK_IS_PROPERTY_OF_V(property::buffer::mem_channel, buffer<double, 3>);
+  CHECK_IS_PROPERTY_OF_V(ext::oneapi::property::buffer::use_pinned_host_memory,
+                         buffer<unsigned int, 2>);
+
+  // Context is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::context::cuda::use_primary_context, context);
+
+  // Image is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::image::use_host_ptr, image<1>);
+  CHECK_IS_PROPERTY_OF_V(property::image::use_mutex, image<2>);
+  CHECK_IS_PROPERTY_OF_V(property::image::context_bound, image<3>);
+
+  // Queue is_property_of_v
+  CHECK_IS_PROPERTY_OF_V(property::queue::in_order, queue);
+  CHECK_IS_PROPERTY_OF_V(property::queue::enable_profiling, queue);
+  CHECK_IS_PROPERTY_OF_V(property::queue::cuda::use_default_stream, queue);
+
+  // Reduction is_property_v
+  CHECK_IS_PROPERTY_V(property::reduction::initialize_to_identity);
```

Comment:
probably should have some negative tests.   `is_property<qux>`  , `is_property_of<buffer, qualia>`   etc.

Reply:
Good idea! I have added negative tests.